lib/descriptor/type/combo.js

```diff
@@ -0,0 +1,128 @@
+/*!
+ * combo.js - combo descriptor object for bcoin
+ * Copyright (c) 2023, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const assert = require('bsert');
+const Descriptor = require('../descriptor');
+const common = require('../common');
+const {isType, strip, checkChecksum, scriptContext, types} = common;
+const KeyProvider = require('../keyprovider');
+const Network = require('../../protocol/network');
+
+/**
+ * ComboDescriptor
+ * Represents a P2PK, P2PKH, P2WPKH, and P2SH-P2WPKH output scripts.
+ * @see https://github.com/bitcoin/bips/blob/master/bip-0384.mediawiki
+ * @property {String} name
+ * @property {KeyProvider[]} keyProviders
+ * @property {Network} network
+ * @extends Descriptor
+ */
+
+class ComboDescriptor extends Descriptor {
+  /**
+   * @constructor
+   * @param {Object} options
+   */
+
+  constructor(options) {
+    super();
+    this.name = types.COMBO;
+
+    if (options) {
+      this.fromOptions(options);
+    }
+  }
+
+  /**
+   * Inject properties from options object.
+   * @param {Object} options
+   * @returns {ComboDescriptor}
+   */
+
+  fromOptions(options) {
+    assert(Array.isArray(options.keyProviders));
+
+    if (options.subdescriptors) {
+      assert(options.subdescriptors.length === 0);
+    }
+
+    assert(
+      options.keyProviders.length === 1,
+      'Can only have one key inside combo()'
+    );
+
+    this.network = Network.get(options.network);
+
+    for (const keyProvider of options.keyProviders) {
```

Comment:
didn't you just assert there will only be one of these? Same with `pk()` maybe others...